robot_smach_states/src/robot_smach_states/navigation/navigate_to_grasp.py

```diff
@@ -6,28 +6,29 @@
 import rospy
 
 # TU/e Robotics
-from cb_planner_msgs_srvs.srv import *
 from cb_planner_msgs_srvs.msg import *
-from robot_skills.arms import Arm
+from robot_skills.arms import PublicArm
 from robot_skills.util.entity import Entity
 from robot_smach_states.navigation import NavigateTo
-from robot_smach_states.util.designators import Designator, check_resolve_type
+from robot_smach_states.util.designators import check_resolve_type
+from robot_smach_states.util.designators.arm import UnoccupiedArmDesignator
 
 
-# ----------------------------------------------------------------------------------------------------
 class NavigateToGrasp(NavigateTo):
     def __init__(self, robot, entity_designator, arm_designator=None):
         super(NavigateToGrasp, self).__init__(robot, reset_head=False)
 
-        self.robot    = robot
+        self.robot = robot
         check_resolve_type(entity_designator, Entity)  # Check that the entity_designator resolves to an Entity
         self.entity_designator = entity_designator
 
-        check_resolve_type(arm_designator, Arm) #Check that the arm_designator resolves to an Arm
+        check_resolve_type(arm_designator, PublicArm)  # Check that the arm_designator resolves to an Arm
+
         self.arm_designator = arm_designator
         if not arm_designator:
-            rospy.logerr('NavigateToGrasp: side should be determined by entity_designator. Please specify left or right, will default to left')
-            self.arm_designator = Designator(robot.leftArm)
+            rospy.logerr('NavigateToGrasp: side should be determined by entity_designator.\
+            Please specify left or right, will default to left. This is Deprecated')
```

Comment:
Why left?

Reply:
As it says. It is deprecated. I don't know what to put in the print. So just added the deprecated sentence.

Reply:
If you know something better. Feel free to change it.